### DISCUSSION TITLE
Many-to-Many Relationship: Unexpected Foreign Key behavior when working with SQLAlchemy

### DISCUSSION THREAD

**Discussion by alv2017** [sqlalchemy/sqlalchemy#12192] [Usage Questions] [answered]
For some reason SQLAlchemy `ignores` Foreign Key constraints. Is is a bug or a feature? :blush:

1) Suppose we created database tables with Foreign Key constraints.
2) Then (by mistake) we attempted to insert some data that violates FK constraints.
3) If we were inserting the violation data directly into DB, then our attempt would fail.
4) However, if we were inserting the violation data using SQLAlchemy, we would succeed. :confused: 

I created a simple script to demonstrate that: https://github.com/alv2017/sqlalchemy-many-to-many-relationship

**Comment by zzzeek** [comment, marked as answer]
SQLite does not enforce FK constraints by default.  see https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#foreign-key-support

**Comment by alv2017** [comment]
Thank you @zzzeek

I didn't know that!  :blush:

I used the "PRAGMA foreign_keys" query, and it turned out that in SQLite Browser the foreign_keys were ON, and in SQLite library used by my Python code foreign_keys were OFF.
